packages/attachments/src/model.ts

```diff
@@ -417,6 +417,16 @@ export class AttachmentsResolver implements IRenderMime.IResolver {
     return true;
   }
 
+  /**
+   * Whether the path is absolute or relative.
+   *
+   * Absolute: /foo/bar, <protocol>://foo/bar, //foo/bar
+   * Relative: ./foo/bar, foo/bar
+   */
+  isAbsolutePath(path: string): boolean {
+    return path[0] === '/' || path.indexOf('//') !== -1 ? true : false;
```

Comment:
Parsing URLs can get pretty tricky. How about this to make it more robust:

1. try using the browser url parsing. If it works, then you have an absolute URL: `try {new URL(path);} catch(e) {/* not an absolute url, but could be an absolute path in a relative URL*/}`

2. If we don't have an absolute URL, then we may still have an absolute path in a relative URL, so just check to see if the first character is `/` (which covers both the cases where it starts with `/` and `//`).

Reply:
Actually, `tel:+1-816-555-1212` is considered an absolute URI, so the strategy above might work still. 

Reading https://tools.ietf.org/html/rfc3986.html#section-4.2, it does seem that we can use the browser `new URL` to determine if we have an absolute or relative URI. If we have a relative URI, we just need to determine if the path is absolute or relative. According to https://tools.ietf.org/html/rfc3986.html#section-4.2, it seems that we can just check the first character - if that first character is a `/`, then the path is absolute, otherwise the path is considered relative (and thus our `/files` should be prepended).

[asdf](//example.com)

Reply:
re #4 - Are there any other cases where the first character is something else?  EG `%`?  As is, new URL("#") is successful, which would make it seem as if it's an absolute path, not a relative path, and thus `/files` would not be prepended.

I can check if the first character is `#` or `?` before the try/catch statement, but want to make sure there aren't any edge cases to content with.

Reply:
I get:

```
> new URL("#")
TypeError: # is not a valid URL.
```

Reply:
FYI, I tried to be very precise in my statements above. There is a difference in the spec between absolute/relative URIs and absolute/relative paths.

Regarding your question about 4, see the spec I linked: https://tools.ietf.org/html/rfc3986.html#section-4.2, namely:

```
      relative-ref  = relative-part [ "?" query ] [ "#" fragment ]
```
If `relative-ref` is the `path-empty` case, it's just a zero-length string, so the relative ref must either start with `?` or `#`.

Reply:
Oh, very interesting. Firefox does not allow `new URL("#")`, but it seems like Chrome does.

Reply:
I looked into the Chrome source code, but it wasn't obvious to me where the behavior is coming from. I think that Chrome may be violating the HTML living spec (https://url.spec.whatwg.org/#api).

Regardless, it looks like this one special case can be checked by looking at the scheme and seeing if it is "about"?

Reply:
[Technical details for future reference]: to be clear, I think Chrome is not following the standard at https://url.spec.whatwg.org/#no-scheme-state. It should start the parsing of the input URL in the [scheme-start-state](https://url.spec.whatwg.org/#scheme-start-state), where case 2 is true, then should skip to [no-scheme-state](https://url.spec.whatwg.org/#no-scheme-state), where it should return an error in step 1 because `base` is null because it wasn't given.